browser/resources/settings/email_aliases_page/email_aliases.tsx

```diff
@@ -0,0 +1,227 @@
+        </MainEmailTextContainer>
+      </Row>
+      <ManageAccountButton
+        title={getLocale('emailAliasesSignOutTitle')}
+        onClick={(e) => { e.preventDefault(); onLogout() }}>
+        <Icon name="outside" />
+        <span>{getLocale('emailAliasesSignOut')}</span>
+      </ManageAccountButton>
+    </AccountRow>
+  </Card>
+
+
+const MainView = ({
+  viewState, mainEmail, onLogout, aliasesState, setViewState,
+  mappingService, onListChange
+}: {
+  viewState: ViewState,
+  mainEmail: string,
+  onLogout: () => void,
+  aliasesState: Alias[],
+  setViewState: (viewState: ViewState) => void,
+  mappingService: MappingService,
+  onListChange: () => void
+}) =>
+  (viewState.mode === 'Startup' ?
+    <Row style={{ margin: '1em', flexGrow: 1, justifyContent: 'center', alignItems: 'center' }}><Icon name='loading-spinner' />
+      <h3>{getLocale('emailAliasesConnectingToBraveAccount')}</h3>
+    </Row> :
+    <span>
+      <MainEmailDisplay onLogout={onLogout} email={mainEmail} />
+      <AliasList aliases={aliasesState} onViewChange={setViewState}
+        mappingService={mappingService}
+        onListChange={onListChange}/>
+    </span>)
+
+export const GrayOverlay = styled.div`
+  background-color: var(--leo-color-dialogs-scrim-background);
+  position: fixed;
+  z-index: 1;
+  inset: 0;
+`
+
+export const ManagePage = ({ mappingService }:
+  {
+    mappingService: MappingService
+  }) => {
+  const [viewState, setViewState] = React.useState<ViewState>({ mode: 'Startup' })
+  const [mainEmail, setMainEmail] = React.useState<string>('')
+  const [aliasesState, setAliasesState] = React.useState<Alias[]>([]);
+  const onEmailChange = async () => {
+    const email = await mappingService.getAccountEmail()
+    setMainEmail(email ?? '')
+    setViewState({ mode: email ? 'Main' : 'SignUp' })
+  }
+  const onListChange = async () => {
+    const aliases = await mappingService.getAliases()
+    setAliasesState(aliases)
+  }
+  const onMainEmailSubmitted = async (email: string) => {
+    setMainEmail(email)
+    await mappingService.requestAccount(email)
+    setViewState({ mode: 'AwaitingAuthorization' })
+    const accountReady = await mappingService.onAccountReady()
+    if (accountReady) {
+      setViewState({ mode: 'Main' })
+      await onListChange()
+    }
+  }
+  const onLogout = () => {
+    mappingService.logout()
+    setViewState({ mode: 'SignUp' })
+  }
+  const onRestart = async () => {
+    await mappingService.cancelAccountRequest()
+    setViewState({ mode: 'SignUp' })
+  }
+  const onReturnToMain = () => {
+    setViewState({ mode: 'Main' })
+    onListChange()
+  }
+  React.useEffect(() => {
+    onEmailChange();
+    onListChange();
+    document.addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'visible') {
+        onEmailChange();
+        onListChange();
+      }
+    })
+    return () => {
+      document.removeEventListener('visibilitychange', () => {})
```

Comment:
nit that won't remove the handler - you have to store it in a variable:

```tsx
const handler = () => {}

document.addEventListener('visibilitychange', handler)
return  () => {
    document.removeEventListener('visibilitychange', handler)
}
```

Reply:
fixed